internal/pkg/container/kernel.go

```diff
@@ -1,60 +1,48 @@
 package container
 
 import (
+	"fmt"
 	"path"
 	"path/filepath"
 	"sort"
 	"strings"
 
+	"github.com/warewulf/warewulf/internal/pkg/kernel"
 	"github.com/warewulf/warewulf/internal/pkg/wwlog"
 )
 
-var (
-	kernelNames = []string{
-		`vmlinux`,
-		`vmlinuz`,
-		`vmlinux-*`,
-		`vmlinuz-*`,
-		`vmlinuz.gz`}
-
-	kernelDirs = []string{
-		`/lib/modules/*/`,
-		`/boot/`}
-)
-
 func KernelFind(container string) string {
```

Comment:
Rather than make `kernelSearchPaths` public, what about creating a new `kernel.Find(root string)` that `container.KernelFind(container string)` calls? This could then be used in `kernel.Build(...)` and `kernel.FindKernelVersion(...)`, reducing code duplication and ensuring they all behave the same.